src/test/java/org/mockitousage/constructor/CreatingMocksWithConstructorTest.java

```diff
@@ -70,7 +115,7 @@ public void exception_message_when_constructor_not_found() {
             fail();
         } catch (MockitoException e) {
             assertThat(e).hasMessage("Unable to create mock instance of type 'HasConstructor'");
-            assertThat(e.getCause()).hasMessageContaining("0-arg constructor");
+            assertThat(e.getCause()).hasMessageContaining("Please ensure that it has a constructor that matches these argument types: []");
```

Comment:
In your PR all unhappy scenarios are handled by generic exception message "Please ensure that it has a constructor that matches these argument types".

I'd like to see fine tuning of the exception message for each scenario because that's how we delight our users! @Brice was working on tuning those exception messages last month, based on feedback/questions from the community. I'd rather not revert his efforts.

I think we have 2 variants:

1. 0-arg constructor (A) or n-arg constructor (B)
2. static/top-level class mocking (X) or inner class mocking (Y)

Here're my suggestions for exception messages:

AX - "Target class does not have 0-arg constructor. Either add 0-arg constructor or specify required constructor arguments."
BX - "Target class does not have matching constructor. Are supplied constructor args correct: []?"
AY - "Is provided outer instance an object of the right type?"
BY - "Target class does not have matching constructor. Is provided outer instance an object of the right type? Are supplied constructor args correct: []?"

In BY, I suggest to get rid of the 1st argument (outer instance type) because most users do not know that java generates that constructor argument automatically.

Above are suggestions - feel free to change it as you work on it.
When we work on fine tuning the exception messages, we write unit tests for scenarios, make them fail, read the messages and asses if they need further tuning, e.g. we check if the exception message accurately describes the scenario and if it informs the user what are the steps to resolve the problem.

Thank you for your patience and improving this PR!!!